docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,301 @@
+
+```bash
+$ kubectl get pods -n openebs -n openebs.io/version=2.2.0
+```
+
+### Step 2: Upgrade OpenEBS data plane components
+
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to `v2.2.0`, the data plane components
+continue to work with the older version `v1.12.0`.
+
+#### Upgrade cStor pools
+
+Get existing `StoragePoolClaims` by executing:
+
+```bash
+kubectl get spc
+NAME                           AGE
+cstor-pool-openebs-replica-3   2d21h26m
+```
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+OPENEBS_NEW_VERSION=2.2.0
+OPENEBS_OLD_VERSION=1.12.0
+cat > upgrade-cstor-pools-1120-220.yaml <<EOF
+# upgrade-cstor-pools-1120-220.yaml
+
+#This is an example YAML for upgrading cstor SPC.
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: cstor-spc-<flattened-from-to-versions>
+  generateName: cstor-spc-1120220-
+  namespace: openebs
+
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+
+        # --from-version is the current version of the pool
+        - "--from-version=${OPENEBS_OLD_VERSION}"
+
+        # --to-version is the version desired upgrade version
+        - "--to-version=${OPENEBS_NEW_VERSION}"
+
+        # Bulk upgrade is supported
+        # To make use of it, please provide the list of SPCs
+        # as mentioned below
+        - "cstor-pool-openebs-replica-3"
+
+        #Following are optional parameters
+        #Log Level
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+
+        # the image version should be same as the --to-version mentioned above
+        # in the args of the job
+        image: openebs/m-upgrade:${OPENEBS_NEW_VERSION}
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+EOF
+```
+
+Apply the job to start the upgrade for cStor pools:
+
+```bash
+kubectl create -f upgrade-cstor-pools-1120-220.yaml
+```
+
+Ensure the job runs to completion.
+
+
+#### Upgrade cStor volumes
+
+Get existing `CStorVolumes` by executing:
+
+```bash
+k get cstorvolumes --all-namespaces
+NAMESPACE   NAME                                       STATUS    AGE     CAPACITY
+openebs     pvc-5b7d8ee3-59b8-4a02-bd6d-a4e66fbecf9f   Healthy   2d21h   50Gi
+openebs     pvc-e8f8b268-b81c-4201-841a-283f557c44a7   Healthy   2d21h   50Gi
+```
+
+
+Create a Kubernetes job to upgrade the existing cStor volumes:
+
+```bash
+OPENEBS_NEW_VERSION=2.2.0
+OPENEBS_OLD_VERSION=1.12.0
```

Comment:
Ditto set at the beginning.

Reply:
I want to set it where I am going to use it in the cat EOF script. It wouldn't make sense at the beginning of the document where its not going to be rendered.

Reply:
I think it's OK to assume they are set and everything happens in the same terminal session. We do it in other tutorials. IMO re-assigning is just confusing.